### ISSUE TITLE
[Bug/Enhancement] Banner Color export on banners with patterns

### ISSUE THREAD

**Issue by XilefTech** [erich666/Mineways#82]
Context: I am a builder at BTE (Build The Earth). I just started making renders of finished builds. That was when I noticed, that all banners would only export as white.
Since we use many banners throughout most of the builts, it really is a shame that the color isn't exported properly and makes it hard to achieve good-looking renders, when 200 orange banners are displayed as white. 

I cannot and won't expect you to be able to export the pattern for every single banner but it would be cool if at least the base color would be exported. I read somewhere on the internet, that that would already be the case for banners without a pattern, yet I haven't tried it. It would be very nice to add these colored banners so that we can create renders that are a lot more realistic to the actual building.

**Comment by erich666** [comment]
It sounds like your project is running on Minecraft 1.12 or earlier - true?

In Mineways 1.13 and newer, if you create solid-colored banners using that object type, Mineways uses such banners:

![image](https://user-images.githubusercontent.com/803995/119265821-1c68bc00-bbb6-11eb-8fdb-eb58f0b2147d.png)

I _think_ I can add some code to the old 1.12 reader to convert to the newer 1.13 format myself, but the code in this area is a bit tricksy.

In the meantime, one workaround is to make a copy and convert your world yourself in Minecraft from 1.12 to 1.13 or newer. The 1.13 or newer world can then be used with Mineways and you'll get banners with their base colors.

**Comment by XilefTech** [comment]
okay, great!
yes, you're right, we're using 1.12
I think, since we are only working with schematics (the entire world would just be way too big) it should be fairly easy to load it in 1.13. I do prefer to use the native version tho so I am looking forward to the update! 

Just a quick side-question: How hard would it actually be to export individual patterns? Would it be possible?
I would love to have the option an I think since BTE is a huge global project, it would be a great contribution to it, if you could make that work! Of course I would understand it, if you don't have enough time to do it.

**Comment by erich666** [comment]
OK, fixed, checked in, but not yet released. However, here's a beta: http://www.realtimerendering.com/erich/minecraft/public/mineways/downloads/Mineways_808_beta_3.zip - let me know if it works OK for you. I've only done basic testing.

> How hard would it actually be to export individual patterns? Would it be possible?

Anything's possible with software :) - well, almost. One headach is that, if you use the "one giant texture" way of exporting worlds (the default), the giant texture has a limited amount of space to it. You can add only so many patterns before it would run out. When using separate textures you could have as many as you like.

Basically, I haven't bothered because it's a pretty involved system for these two block types (standing and wall). It would also be involved, in that Mineways would have to read all the pattern textures for the resource pack and store these away, etc. Mineways has lots of limitations, no signs with letters, paintings go away, heads are always pumpkins, on and on - it's a hobby.

**Comment by erich666** [comment]
Speaking of hobbies, keep up the nice work on BTE! Definitely send me a link to what renders you make, @XilefTech.

**Comment by XilefTech** [comment]
sure, do you have discord? or what would be your preferred way of communication?

**Comment by XilefTech** [comment]
btw. the updated version worked well so far! Thank you very much!

**Comment by erich666** [comment]
Ah, good to hear it's working - thanks!

> sure, do you have discord? or what would be your preferred way of communication?

The Mineway discord server is https://discord.gg/wfayQA6Dyc

But, really, I respond best to email: erich@acm.org (and I'm not afraid to put my email - Google's spam filter works well nowadays).
